work_queue/src/work_queue_worker.c

```diff
@@ -1931,6 +1931,33 @@ static int serve_master_by_hostport( const char *host, int port, const char *ver
 
 	measure_worker_resources();
 
+	//Check GPU name if we can
+	char *gpu_name;	
+	int fd = open("/bin/nvidia-smi", O_RDONLY);
```

Comment:
Nate,

Please see   dttools/src/cctools_gpu_autodetect

Is that tool doing the same things? If so, please consolidate (remove the script, or remove this c code).